src/coordinated_workers/coordinator.py

```diff
@@ -469,19 +488,154 @@ def _reconcile(self):
     # UTILITY PROPERTIES #
     ######################
 
+    def _tracing_receivers_urls(
+        self,
+        requirer: TracingEndpointRequirer,
+        _type: str,
+        ignore_proxy: bool = False,
+    ) -> Dict[str, str]:
+        """Returns the trace receiving urls per requested protocol."""
+        endpoints = requirer.get_all_endpoints()  # type: ignore
+        receivers = endpoints.receivers if endpoints else ()
+
+        if (proxy_worker_telemetry_port := self._proxy_worker_telemetry_port) and not ignore_proxy:
+            return worker_telemetry.proxy_tracing_receivers_urls(
+                hostname=self.app_hostname(
+                    self.hostname, self._charm.app.name, self._charm.model.name
+                ),
+                proxy_worker_telemetry_port=proxy_worker_telemetry_port,
+                tls_available=self.tls_available,
+                tracing_target_type=_type,
+                protocols=[receiver.protocol.name for receiver in receivers],  # type: ignore
+            )
+
+        return {receiver.protocol.name: receiver.url for receiver in receivers}  # type: ignore
+
     @property
     def _charm_tracing_receivers_urls(self) -> Dict[str, str]:
-        """Returns the charm tracing enabled receivers with their corresponding endpoints."""
-        endpoints = self.charm_tracing.get_all_endpoints()
-        receivers = endpoints.receivers if endpoints else ()
-        return {receiver.protocol.name: receiver.url for receiver in receivers}
+        """Returns the charm tracing enabled receivers per tracing protocol with their corresponding endpoints to be published into the cluster data.
+
+        When worker telemetry proxy is enabled and when not self tracing, returns the proxy url for sending traces.
+
```

Comment:
this docstring makes no sense to me.
Can you rephrase to make it more clear? Same below.